plugins/modules/k8s_taint.py

```diff
@@ -0,0 +1,279 @@
+        self.k8s_ansible_mixin.params = self.module.params
+        self.k8s_ansible_mixin.fail_json = self.module.fail_json
+        self.k8s_ansible_mixin.fail = self.module.fail_json
+        self.k8s_ansible_mixin.exit_json = self.module.exit_json
+        self.k8s_ansible_mixin.warn = self.module.warn
+        self.k8s_ansible_mixin.warnings = []
+        self.api_instance = core_v1_api.CoreV1Api(self.k8s_ansible_mixin.client.client)
+        self.k8s_ansible_mixin.check_library_version()
+        self.changed = False
+
+    def get_node(self, name):
+        try:
+            node = self.api_instance.read_node(name=name)
+        except ApiException as exc:
+            if exc.reason == "Not Found":
+                self.module.fail_json(msg="Node '{0}' has not been found.".format(name))
+            self.module.fail_json(
+                msg="Failed to retrieve node '{0}' due to: {1}".format(
+                    name, exc.reason
+                ),
+                status=exc.status,
+            )
+        except Exception as exc:
+            self.module.fail_json(
+                msg="Failed to retrieve node '{0}' due to: {1}".format(
+                    name, to_native(exc)
+                )
+            )
+
+        return node
+
+    def patch_node(self, taints):
+        body = {"spec": {"taints": taints}}
+
+        try:
+            result = self.api_instance.patch_node(
+                name=self.module.params.get("name"), body=body
+            )
+        except Exception as exc:
+            self.module.fail_json(
+                msg="Failed to patch node due to: {0}".format(to_native(exc))
+            )
+
+        return result.to_dict()
+
+    def execute_module(self):
+        result = {"result": {}}
+        state = self.module.params.get("state")
+        taints = self.module.params.get("taints")
+        name = self.module.params.get("name")
+
+        node = self.get_node(name)
+        existing_taints = node.spec.to_dict()["taints"] or []
+
+        diff = _get_difference(taints, existing_taints)
+
+        if state == "present":
+            if not diff:
+                result["result"] = node.to_dict()
+                self.module.exit_json(changed=self.changed, **result)
+            else:
+                if self.module.check_mode:
+                    self.changed = True
+                    self.module.exit_json(changed=self.changed, **result)
+
+                if not self.module.params.get("overwrite"):
+                    result["result"] = self.patch_node(
+                        taints=[*existing_taints, *taints]
+                    )
+                else:
+                    result["result"] = self.patch_node(taints=taints)
+
+                self.changed = True
+
+        elif state == "absent":
+            if not existing_taints:
+                result["result"] = node.to_dict()
+            if not diff:
+                self.changed = True
```

Comment:
module execution should not reflect changes when using ``check_mode``

Reply:
In `check_mode` we don't trigger any change, but we only simulate it.